src/main/java/io/vertx/core/http/impl/HttpHandlers.java

```diff
@@ -41,6 +41,25 @@ public HttpHandlers(
     this.exceptionHandler = exceptionHandler;
   }
 
+  Handler<HttpServerRequest> requestHandler() {
```

Comment:
Having this logic here is confusing. For HTTP/1 this method is invoked when the server connection is created. For HTTP/2, the raw `requestHandler` is read.

`HttpHandlers` should remain a data class and this logic be moved to the `Http1xServerConnection` constructor.

Reply:
sounds a better thing to do indeed.